### DISCUSSION TITLE
styling maps

### DISCUSSION THREAD

**Discussion by humanitiesNerd** [protomaps/PMTiles#307] [General]
I rendered this map using a style I found named "OSM Bright"

![pmtiles2](https://github.com/protomaps/PMTiles/assets/283791/5b41a313-4365-4a1e-918a-727d664a8d97)

It's not bad but the style on the officlal openstreetmap site is way more detailed and expressive

I took a look at Carto but it's extremely complex and assumes a ridicolously complicated setup

And then, there are some more efforts, around on the net, of map renderings

I was wondering if there's a style I could use that renders more and more details

Or if there's a common effort for a style that can be used by small projects like mine

**Comment by bdon** [comment]
The official OSM style "OpenStreetMap Carto" is detailed and expressive because the setup is very complicated - it's designed for near real-time viewing of almost 100% of data in OSM. The basemaps design here https://github.com/protomaps/basemaps is designed for a reasonable and opinionated tradeoff between detail and tile size+complexity, similar to Google or Apple Maps. Do you have a specific set of features you want visualized?

**Reply by humanitiesNerd** [reply]
My idea was to have roughly the same of what Carto has, but for the "near real time" part

Meaning, my assumption was that I'd have produced the tiles of the area I'm interested in in a batch fashion, manually

And then I'd have had almost everything Carto has

I hadn't realized that the repo you linked was meant for this and that a basemap was available at all, so thanks for that

I'll try that and see where it gets me

Thanks

**Comment by humanitiesNerd** [comment]
Ok, it runs

![immagine](https://github.com/protomaps/PMTiles/assets/283791/6f8d8053-649f-4411-8cdc-26e21f13f6b4)


but how do I get s json file for the style ?

npm run build failes with

```
$ npm run build

> protomaps-basemaps-app@0.0.0 build
> tsc && vite build

../styles/src/base_layers.ts:2:36 - error TS2307: Cannot find module '@maplibre/maplibre-gl-style-spec' or its corresponding type declarations.

2 import { LayerSpecification } from "@maplibre/maplibre-gl-style-spec";
                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

../styles/src/index.ts:3:36 - error TS2307: Cannot find module '@maplibre/maplibre-gl-style-spec' or its corresponding type declarations.

3 import { LayerSpecification } from "@maplibre/maplibre-gl-style-spec";
                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 2 errors in 2 files.

Errors  Files
     1  ../styles/src/base_layers.ts:2
     1  ../styles/src/index.ts:3
```

**Reply by bdon** [reply]
press the "get style json" button at the top to see the JSON style.

**Comment by wipfli** [comment]
You might want to have a look at Apache Baremaps @humanitiesNerd. See https://baremaps.apache.org/
